packages/connect-core/src/transform-iterable.ts

```diff
@@ -40,166 +40,368 @@ import { compressedFlag, Compression } from "./compression.js";
+  transform5: AsyncIterableTransform<T5, T6>,
+  sink: AsyncIterableSink<T6, T7>,
+  options?: PipeOptions
+): Promise<T7>;
+/**
+ * Takes an asynchronous iterable as an input, applies transformations, and
+ * passes it to a sink.
+ */
+export function pipeTo<T1, T2, T3, T4, T5, T6, T7, T8>(
+  iterable: AsyncIterable<T1>,
+  transform1: AsyncIterableTransform<T1, T2>,
+  transform2: AsyncIterableTransform<T2, T3>,
+  transform3: AsyncIterableTransform<T3, T4>,
+  transform4: AsyncIterableTransform<T4, T5>,
+  transform5: AsyncIterableTransform<T5, T6>,
+  transform6: AsyncIterableTransform<T6, T7>,
+  sink: AsyncIterableSink<T7, T8>,
+  options?: PipeOptions
+): Promise<T8>;
+/**
+ * Takes an asynchronous iterable as an input, applies transformations, and
+ * passes it to a sink.
+ */
+export function pipeTo<T1, T2, T3, T4, T5, T6, T7, T8, T9>(
+  iterable: AsyncIterable<T1>,
+  transform1: AsyncIterableTransform<T1, T2>,
+  transform2: AsyncIterableTransform<T2, T3>,
+  transform3: AsyncIterableTransform<T3, T4>,
+  transform4: AsyncIterableTransform<T4, T5>,
+  transform5: AsyncIterableTransform<T5, T6>,
+  transform6: AsyncIterableTransform<T6, T7>,
+  transform7: AsyncIterableTransform<T7, T8>,
+  sink: AsyncIterableSink<T8, T9>,
+  options?: PipeOptions
+): Promise<T9>;
+/**
+ * Takes an asynchronous iterable as an input, applies transformations, and
+ * passes it to a sink.
+ */
+export function pipeTo<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10>(
+  iterable: AsyncIterable<T1>,
+  transform1: AsyncIterableTransform<T1, T2>,
+  transform2: AsyncIterableTransform<T2, T3>,
+  transform3: AsyncIterableTransform<T3, T4>,
+  transform4: AsyncIterableTransform<T4, T5>,
+  transform5: AsyncIterableTransform<T5, T6>,
+  transform6: AsyncIterableTransform<T6, T7>,
+  transform7: AsyncIterableTransform<T7, T8>,
+  transform8: AsyncIterableTransform<T8, T9>,
+  sink: AsyncIterableSink<T9, T10>,
+  options?: PipeOptions
+): Promise<T10>;
+// prettier-ignore
+/**
+ * Takes an asynchronous iterable as an input, applies transformations, and
+ * passes it to a sink.
+ */
+export function pipeTo<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11>(
+  iterable: AsyncIterable<T1>,
+  transform1: AsyncIterableTransform<T1, T2>,
+  transform2: AsyncIterableTransform<T2, T3>,
+  transform3: AsyncIterableTransform<T3, T4>,
+  transform4: AsyncIterableTransform<T4, T5>,
+  transform5: AsyncIterableTransform<T5, T6>,
+  transform6: AsyncIterableTransform<T6, T7>,
+  transform7: AsyncIterableTransform<T7, T8>,
+  transform8: AsyncIterableTransform<T8, T9>,
+  transform9: AsyncIterableTransform<T9, T10>,
+  sink: AsyncIterableSink<T10, T11>,
+  options?: PipeOptions
+): Promise<T11>;
+// prettier-ignore
+/**
+ * Takes an asynchronous iterable as an input, applies transformations, and
+ * passes it to a sink.
+ */
+export function pipeTo<T1, T2, T3, T4, T5, T6, T7, T8, T9, T10, T11, T12>(
+  iterable: AsyncIterable<T1>,
+  transform1: AsyncIterableTransform<T1, T2>,
```

Comment:
Would be cool to encapsulate these somehow into a class or entity like a WHATWG pipe chain called `AsyncIterableTransformChain` or something that had all these overloads inside, which would allow these signatures to be cleaned up a bit to something like:

```ts
export function pipeTo(
  iterable: AsyncIterable<I>,
  transformChain: AsyncIterableTransformChain<I, O>,
  sink: AsyncIterableSink<O, T>,
  options?: PipeOptions
): Promise<T>;
```
No clue at all if that would work, but just spitballing.